### ISSUE TITLE
Advanced Numeric Filter does not register input in special case

### ISSUE THREAD

**Issue by kerajel** [radzenhq/radzen-blazor#1809]
The issue is reproduced only on hosted environment (that is, not reproduced when the application runs locally in debug e.g. launched from Visual Studio)
The issue is reproduced only on numeric advanced filter (integers, decimals alike)

Preconditions
RadzenDataGrid with LoadData and advanced filtering
The LoadData has a dialog service that shows current LoadDataArgs's filter:

```c#
public override async Task LoadData(LoadDataArgs loadDataArgs)
{
	await DialogService.Alert($"{loadDataArgs.Filter}");
	
	entries = await GetEntries();
	totalCount = await GetTotalCount();
}
```

When I enter a numeric input and press 'Apply' right away then my input is not registered in the LoadDataArgs.Filter. Here is a gif that showcases that:
![NoClickAway](https://github.com/user-attachments/assets/1e924cdc-46bd-46ea-a3fc-be32c61f2e7f)

But when I enter the same numeric input and click away from the input (e.g. on the area near the second filter) then my input is registered in LoadDataArgs.Filter. Here is a gif that showcases that:
![ClickAway](https://github.com/user-attachments/assets/58073ff3-cf0c-4904-909a-562d59537589)

I have not observed such behavior around 2 weeks ago, so it must be related to the recent changes. Access to the website where this is observed can be granted if it helps.

.NET 9.0
Radzen.Blazor Version="5.6.7"
Google Chrome Version 131.0.6778.86 (Official Build) (64-bit)
Hosting: SmarterAsp.NET

**Comment by enchev** [comment]
We cannot provide any advice by looking at a gif. Use our source code to debug your app.

**Comment by kerajel** [comment]
@enchev my app is fine, it's the numeric filter component that requires debugging.

I have skimmed last weeks' commits and found this particular one: https://github.com/radzenhq/radzen-blazor/commit/124033d122ba981f554498c3b8cefdd9a0bbe9e7#diff-7ad12b64ebe222b4a5a33c2735cbd89d58bac88807d49ffb3c5a23ffb41357a8L818

In scope of this commit the oninput handler was removed. By removing this event handler, the numeric input field now relies on the default onchange event, which only fires when the input loses focus, matching my observations.

I can additionally confirm that this commit broke the input by downgrading to 5.6.0 - before the commit - the issue is not observed on this version.

Although removing of the handler introduced the current problem it has fixed the previous issue I had reported (which was similarly neglected and closed without due investigation)
https://github.com/radzenhq/radzen-blazor/issues/1787
